### ISSUE TITLE
Line-break in text editor is not working after reload

### ISSUE THREAD

**Issue by NykolaySavin** [tinymce/tinymce-react#216]
We have controlled editor with text output format. When using value and onEditorChange props - line breaks are not shown.

Bug is shown in the codesandbox:
https://codesandbox.io/s/tinymcereact-forked-h2kdm?file=/src/index.js 

We expect line-breaks to be good generated with the help of value.

**Comment by tiny-james** [comment]
The problem is that the value prop always expects HTML. As whitespace is collapsed in HTML the line breaks become a single space.

I admit that the `outputFormat` prop is not very useful because of this limitation however generally attempting to convert from text to HTML automatically (and vice-versa) gets inconsistent results so I recommend avoiding it if you can.

If you only have to set the HTML from the text once at the start then you could approximate the text layout as follows:

1. Encode any special characters (ie `<`, `>`, `"`, `'` and `&`). I'd recommend doing this  by setting the `textContent` of a div and then getting the `innerHTML`.
2. Convert runs of multiple spaces into a space followed by non-breaking spaces, also convert any spaces at the start or end of lines into nbsp.
3. Split into parts separated by two or more newlines or carriage returns
4. Convert individual newlines (or carriage returns) inside the parts into `<br/>` tags
5. Wrap the parts in `<p>` and `</p>` tags
6. Join the parts back into one string

Note that the text you get out of the editor may be slightly different from the text that you put in. I know from experience trying to get round-tripping working in another project that it is essentially impossible to get identical results.

Good luck

**Comment by tiny-james** [comment]
I created a PR that implements what you requested though I am undecided if it is a good idea:
https://github.com/tinymce/tinymce-react/pull/222

To decide the best course of action we would like to know how you intend to make use of `outputFormat="text"` and what you aim to achieve - also why a textarea is not suitable for your use case.

**Comment by tiny-james** [comment]
Closing this as "out of scope". 

Converting from text to HTML must be done before passing to the editor as the editor only handles HTML no matter what the `outputFormat` is set to. The `outputFormat` prop has been marked as deprecated in the 4.11 release because it leads to these confusing situations. If the text output is still required it can be retrieved by calling `editor.getContent({format: 'text'})` from any of the event handlers.
